src/engine/shared.rs

```diff
@@ -162,3 +162,116 @@ pub fn set_key_shared(key_fd: RawFd, interactive: Option<bool>) -> StratisResult
 
     Ok(sized_memory)
 }
+
+/// Validate a str for use as a Pool or Filesystem name.
+pub fn validate_name(name: &str) -> StratisResult<()> {
+    let name_path = Path::new(name);
+    if name.contains('\u{0}') {
+        return Err(StratisError::Engine(
+            ErrorEnum::Invalid,
+            format!("Name contains NULL characters : {}", name),
+        ));
+    }
```

Comment:
Is there a reason that this doesn't come before `let name_path ...`?

Reply:
I doubt it! I didn't look inside this method when I moved it. I'll swap the order.

Reply:
Perfect, thanks!